fenix/app/src/main/java/org/mozilla/fenix/debugsettings/data/DebugSettingsCache.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.debugsettings.data
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+/**
+ * [DataStore] for accessing debugging settings.
+ */
+private val Context.debugSettings: DataStore<Preferences> by preferencesDataStore(name = "debug_settings")
+
+private val debugDrawerEnabledKey = booleanPreferencesKey("debug_drawer_enabled")
+
+/**
+ * Cache for accessing any settings related to debugging.
+ */
+interface DebugSettingsCache {
+
+    /**
+     * [Flow] for checking whether the Debug Drawer is enabled.
+     */
+    val debugDrawerEnabledFlow: Flow<Boolean>
+
+    /**
+     * Updates whether the debug drawer is enabled.
+     *
+     * @param enabled Whether the debug drawer is enabled.
+     */
+    suspend fun setDebugDrawerEnabled(enabled: Boolean)
+}
+
+/**
+ * The default implementation of [DebugSettingsCache].
+ *
+ * @param context Android context used to obtain the underlying [DataStore].
+ * @param writeScope [CoroutineScope] used for writing settings changes to disk.
+ */
+class DefaultDebugSettingsCache(
+    context: Context,
+    private val writeScope: CoroutineScope,
+) : DebugSettingsCache {
+
+    private val dataStore = context.debugSettings
+
+    override val debugDrawerEnabledFlow: Flow<Boolean> =
+        dataStore.data.map { preferences ->
+            preferences[debugDrawerEnabledKey] ?: false
+        }
+
+    override suspend fun setDebugDrawerEnabled(enabled: Boolean) {
+        writeScope.launch {
+            dataStore.edit { preferences ->
+                preferences[debugDrawerEnabledKey] = enabled
+            }
+        }
+    }
+
+    internal fun clearPreferences() {
```

Comment:
It might be better to do a private function that's visible for test until this lives in a separate module.

Reply:
??? Making it private will hide it from the tests. Adding the usual "@VisibleForTesting" simply adds the syntax squiggle underline when used otherwise. I'll add the annotation for the highlight as well as a comment.

Reply:
Let's pass in the `dataStore` so it can be cleared in the test file then this isn't required 🙂 Again similar to `ReportSiteDomainsRepositoryTest`

Reply:
that's fair. I'll look into this